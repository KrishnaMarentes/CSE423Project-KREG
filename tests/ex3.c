int foo(int k, float m) {
    int tmp = k - m;
    return (tmp << 2) * 4;
}

int main(int argc, char** argv)
{
    int i,j=3, m=4,f;

    while(i < 2 || 2 == 2) {
        i++;
        {
            break;
            t--;
        }
    }

    return y;
}