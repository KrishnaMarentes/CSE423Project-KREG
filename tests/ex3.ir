function foo(int m,int f)
{
int KREG.2;
KREG.2 = m + f;
return KREG.2;
}

function main(int argc,char* argv)
{
int m;
int a;
float t;
float f;
int unreachable;
int KREG.3;
int KREG.4;
int KREG.5;
int KREG.6;
int KREG.7;
int KREG.8;
int KREG.9;
int KREG.10;
int KREG.11;
int KREG.12;
int KREG.13;
int KREG.14;
int KREG.15;
int KREG.16;
goto <KREG.1>;
KREG.3 = 10 * 10;
unreachable = KREG.3;
<KREG.1>:
a = a + 1;
KREG.5 = a;
a = a + 1;
KREG.6 = a;
KREG.7 = foo();
KREG.8 = foo(KREG.7);
KREG.9 = KREG.8 * 2;
m = KREG.9;
KREG.10 = KREG.9 * a;
a = KREG.10;
KREG.11 = KREG.9 * 5;
KREG.12 = foo(KREG.11,f);
KREG.13 = KREG.9 * KREG.12;
t = KREG.13;
KREG.14 = KREG.9 + 4;
KREG.15 = 594;
KREG.16 = KREG.14 + 594;
return KREG.16;
}


