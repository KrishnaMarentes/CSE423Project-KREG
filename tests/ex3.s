.section .text

.global foo
foo:
	push %ebp
	mov %esp, %ebp
	sub $4, %esp
	mov 8(%ebp), %ecx
	mov 12(%ebp), %edx
	add %ecx, %edx
	mov %edx, %eax
	leave
	ret
.global main
main:
	push %ebp
	mov %esp, %ebp
	sub $68, %esp
	jmp KREG.1
	mov $10, %ecx
	mov $10, %edx
	imul %ecx, %edx
	mov %edx, -12(%ebp)
KREG.1:
	mov -8(%ebp), %ecx
	mov $1, %edx
	add %ecx, %edx
	mov %edx, -8(%ebp)
	mov -8(%ebp), %ebx
	mov %ebx, -24(%ebp)
	mov -8(%ebp), %ecx
	mov $1, %edx
	add %ecx, %edx
	mov %edx, -8(%ebp)
	mov -8(%ebp), %ebx
	mov %ebx, -28(%ebp)
	push $
	call foo
	mov %eax, -32(%ebp)
	push -32(%ebp)
	call foo
	mov %eax, %ecx
	mov $2, %edx
	imul %ecx, %edx
	mov %edx, -4(%ebp)
	mov -40(%ebp), %ecx
	mov -8(%ebp), %edx
	imul %ecx, %edx
	mov %edx, -8(%ebp)
	mov -40(%ebp), %ecx
	mov $5, %edx
	imul %ecx, %edx
	mov %edx, -48(%ebp)
	push $f
	push -48(%ebp)
	call foo
	mov %eax, -52(%ebp)
	mov -40(%ebp), %ecx
	mov -52(%ebp), %edx
	imul %ecx, %edx
	mov %edx, $t
	mov -40(%ebp), %ecx
	mov $4, %edx
	add %ecx, %edx
	mov %edx, -60(%ebp)
	movl $594, -64(%ebp)
	mov -60(%ebp), %ecx
	mov $594, %edx
	add %ecx, %edx
	mov %edx, %eax
	leave
	ret
