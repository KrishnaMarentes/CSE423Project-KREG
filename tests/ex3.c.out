program 
  declarationList 
    declaration 
      funDeclaration 
        typeSpecifier int
        main(params)
        compoundStmt {
          statementList statementList
            statement 
              returnStmt return
                expression 
                  simpleExpression 
                    andExpression 
                      unaryRelExpression 
                        relExpression 
                          sumExpression 
                            mulExpression 
                              mulExpression 
                                unaryExpression 
                                  factor 
                                    immutable 
                                      constant 1
                              mulop *
                              unaryExpression 
                                factor 
                                  immutable 
                                    constant 2
                ;
          }
  <EOF>

program 
  funDeclaration 
    typeSpecifier int
    main
    returnStmt return
      mulExpression 
        constant 1
        mulop *
        constant 2


